### ISSUE TITLE
Arbitrary-ish helical paths

### ISSUE THREAD

**Issue by altaic** [CadQuery/cadquery#984]
I'm trying to create helical paths around shapes other than a cylinder or cone. For example, I made this manually by splitting a helix and drawing tangent lines between the two halves:
<img width="996" alt="Screen Shot 2022-02-06 at 11 16 38 PM" src="https://user-images.githubusercontent.com/146901/152726012-95e2f4a5-5385-4a5e-a5fb-10f25d3a18bd.png">

However, I'd like to be able to do this with any closed ruled surface, or more specifically if I have a closed 2D curve on a sketch, I want to create a helix that follows that curve.

I saw that `makeHelix` works by mapping an angled line onto a surface (cylinder or cone) via u v coordinates. That sounds pretty good to me, but I'm getting lost in wrapped types and internal geometry representations. I'd like to use CQ's nice high level API if possible.

Alternatively, if it can be calculated directly with splines, I'd appreciate any pointers to papers about that. That said, I haven't gone very far down the white paper rabbit hole since I'm hoping there's an easier way.

Thanks!

**Comment by adam-urbanczyk** [comment]
I don't think that it is implemented. The best would be to follow the approach of `makeHelix`. You could also try `cq.Workplane.parametricCurve` but you'd need to know the parametric equations of you helix.

**Comment by altaic** [comment]
Thanks, I'll check out `parametricCurve` and figure out what's sensible from white papers.

**Comment by altaic** [comment]
I got it working using `parametricCurve`, however I am looking for good alternatives. `parametricCurve` is extremely fragile and requires tweaking of `N` and `tol`, and even then it may not yield good results. In all of my tests, `N` had to be significantly lower (e.g. `N=100`) than the default (`N=400`) to get a result that wasn't completely broken.
